src/shared-components/segmentedControl/index.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useRef, useEffect } from 'react';
+
+import { SegmentsContainer, SegmentItem, Indicator } from './style';
+import { SegmentedControlProps, SegmentItemType } from './types';
+
+/**
+ * The width of the top-level container is set to 100%, so it will expand to fill its parent container. Set a non-percentage `width` on the parent element during implementation to avoid stretched-out layout or animation effects.
+ */
+export const SegmentedControl: React.FC<SegmentedControlProps> = ({
+  segmentItems,
+  initialActiveId = 1,
+  onClick,
+}) => {
+  if (segmentItems.length === 0) {
+    return null;
+  }
+  const initialActiveItem = segmentItems.find(
+    (item: SegmentItemType) => item.id === initialActiveId,
+  );
+  const initialActiveIndex = initialActiveItem
+    ? segmentItems.indexOf(initialActiveItem)
+    : 0;
+
+  const [activeSegmentText, setActiveSegmentText] = useState(() =>
+    initialActiveItem ? initialActiveItem.text : segmentItems[0].text,
+  );
+  const [activeSegmentIndex, setActiveSegmentIndex] = useState(
+    initialActiveIndex,
+  );
+  const targetRef = useRef<HTMLButtonElement>(null);
+  const [transform, setTransform] = useState('');
+  const [targetWidth, setTargetWidth] = useState(0);
+  const segmentWidth = 100 / segmentItems.length;
+
+  useEffect(() => {
+    if (targetRef.current) {
+      setTargetWidth(targetRef.current.offsetWidth);
+      setTransform(
+        `translate3d(${
+          targetRef.current.offsetWidth * activeSegmentIndex
+        }px, 0, 0)`,
+      );
+    }
+  }, [targetRef]);
+
+  const onSegmentClick = (segment: SegmentItemType, index: number) => {
+    setActiveSegmentText(segment.text);
+    setActiveSegmentIndex(index);
+    setTransform(`translate3d(${targetWidth * index}px, 0, 0)`);
+    return onClick ? onClick(segment) : null;
+  };
+
+  return (
+    <SegmentsContainer>
+      <Indicator width={segmentWidth} transform={transform} ref={targetRef}>
```

Comment:
Is there a way to consolidate `Indicator` within the `segmentItems` map function to prevent adding an additional button to the DOM? It looks like there's always one more than there exists `segmentItems`:

<img width="972" alt="Screen Shot 2020-10-20 at 7 53 45 PM" src="https://user-images.githubusercontent.com/13544620/96659972-3cbd5900-12fd-11eb-9871-d0e2ed806b05.png">

Also, since Indicator precedes all the segment items, the current keyboard navigation behave is to immediately tab to the currently active segment. Is this intended? E.g., if Option 2 of 3 options is active, tabbing will go to 2, then to 1, then to 3. My expectation is that it would go in order.